server/index.js

```diff
@@ -1,104 +1,151 @@
-app.use(categories)
-app.use(playlists)
+  app.use(pages)
+  app.use(cache)
 
-postload.forEach((middleware) => app.use(middleware))
+  // category pages will be cache busted when their last updated timestamp changes
+  app.use(categories)
+  app.use(playlists)
 
-// if no page has been served, check for a redirect before erroring
-app.use(redirects)
+  postload.forEach((middleware) => app.use(middleware))
 
-// error handler for rendering the 404 and 500 pages, must go last
-app.use(errorPages)
+  // if no page has been served, check for a redirect before erroring
+  app.use(redirects)
 
-// If we are called directly, listen on port 3000, otherwise don't
+  // error handler for rendering the 404 and 500 pages, must go last
+  app.use(errorPages)
 
-if (require.main === module) {
-  app.listen(parseInt(process.env.PORT || '3000', 10))
+  // If we are called directly, listen on port 3000, otherwise don't
+
+  if (require.main === module) {
+    app.listen(parseInt(process.env.PORT || '3000', 10))
+  }
+
+  module.exports = app
 }
 
-module.exports = app
+(async () => {
+  const gcpMetadata = require('gcp-metadata');
+  const { SecretManagerServiceClient } = require('@google-cloud/secret-manager');
+
+  // GCP Metadata is only available to the deployed instance.
+  // Running locally, you would still use the .env file.
+  const isDeployedInstance = await gcpMetadata.isAvailable();
+
+  if (isDeployedInstance) {
+    // We're on GCP here.  Load up environment variables from 
+    // Secrets Manager
+    const projectId = await gcpMetadata.project('project-id');
+
+    // Get Variable
+    const client = new SecretManagerServiceClient()
+
+    process.env.APPROVED_DOMAINS = await getSecret(client, projectId, 'APPROVED_DOMAINS', false);
+    process.env.DRIVE_ID = await getSecret(client, projectId, 'DRIVE_ID', false);
+    process.env.DRIVE_TYPE = await getSecret(client, projectId, 'DRIVE_TYPE', false);
+    process.env.EXCLUDE_FOLDERS = await getSecret(client, projectId, 'EXCLUDE_FOLDERS', false);
+    process.env.GCP_PROJECT_ID = await getSecret(client, projectId, 'GCP_PROJECT_ID', false);
+    process.env.NODE_ENV = await getSecret(client, projectId, 'NODE_ENV', false);
+    process.env.REDIRECT_URL = await getSecret(client, projectId, 'REDIRECT_URL', false);
+    process.env.GOOGLE_APPLICATION_JSON = await getSecret(client, projectId, 'GOOGLE_APPLICATION_JSON', true);
+    process.env.GOOGLE_APPLICATION_CREDENTIALS = await getSecret(client, projectId, 'GOOGLE_APPLICATION_CREDENTIALS', false);
+    process.env.GOOGLE_APPLICATION_PUBLIC_KEY = await getSecret(client, projectId, 'GOOGLE_APPLICATION_PUBLIC_KEY', true);
+    process.env.GOOGLE_CLIENT_ID = await getSecret(client, projectId, 'GOOGLE_CLIENT_ID', false);
+    process.env.GOOGLE_CLIENT_SECRET = await getSecret(client, projectId, 'GOOGLE_CLIENT_SECRET', false);
+    process.env.SESSION_SECRET = await getSecret(client, projectId, 'SESSION_SECRET', false);
```

Comment:
instead of getting each of these, can we do a `listSecrets` call and then just pull all of those into our env? I think we do something similar on People App startup

Reply:
Yep!  I agree, `loadSecrets` way is better.